### PR TITLE
[CARBONDATA-222]  Handled query issue for all dimensions are no dictionary columns

### PR DESCRIPTION
**Pull request by foryou2030**
# Why raise this pr?
query failed when all dimensions are no-dictionary column, and the data contains one row that all no-dictionary column value are empty
# How to resolve?
when find First Data Block, we need to compare indexes, in this scenario,we should use other compareTo method.

### CODE REVIEW

**Comment by foryou2030** [comment]
CI passed
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/135/

File: core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java [hunk 1]
```diff
@@ -240,9 +240,16 @@ private int compareIndexes(IndexKey first, IndexKey second) {
               firstNoDictionaryKeyBuffer.getShort(nonDictionaryKeyOffset + SHORT_SIZE_IN_BYTES);
           secondNodeDictionaryLength =
               secondNoDictionaryKeyBuffer.getShort(nonDictionaryKeyOffset + SHORT_SIZE_IN_BYTES);
-          compareResult = ByteUtil.UnsafeComparer.INSTANCE
-              .compareTo(first.getNoDictionaryKeys(), actualOffset, firstNoDcitionaryLength,
-                  second.getNoDictionaryKeys(), actualOffset, secondNodeDictionaryLength);
+          int minLength = Math.min(firstNoDcitionaryLength, secondNodeDictionaryLength);
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java]: @kumarvishal09 please comment on this. I am not getting how actualOffset is same for both keybuffers. 
Also as I know length can be calculated by nextOffset-currentOffset. Please give the structure of data stored and how this function works

Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java]: @gvramana Yes you right current logic has some problem we need to maintain offset of both the keys and when it is not the last key we need to substract the secondNodeDictionaryLength - actualOffset1 and firstNodeDictionaryLength-actualOffset.

**Comment by foryou2030** [comment]
CI passed:
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/145/

**Comment by kumarvishal09** [comment]
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/145/

**Comment by kumarvishal09** [comment]
LGTM

File: core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java (hunk 1 above)
Comment (foryou2030) [inline core/src/main/java/org/apache/carbondata/core/carbon/datastore/impl/btree/BTreeDataRefNodeFinder.java]: ok, handled. thanks for you @gvramana @kumarvishal09

**Pull request closed without merging**